src/clients/c++/image_client.cc

```diff
@@ -43,10 +43,17 @@
 #include <opencv2/core/core.hpp>
 #include <opencv2/highgui/highgui.hpp>
 #include <opencv2/imgproc/imgproc.hpp>
+#define GET_TRANSFORMATION_CODE(x) CV_##x
 #elif CV_MAJOR_VERSION == 3
 #include <opencv2/core.hpp>
 #include <opencv2/highgui.hpp>
 #include <opencv2/imgproc.hpp>
+#define GET_TRANSFORMATION_CODE(x) CV_##x
+#elif CV_MAJOR_VERSION == 4
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+#define GET_TRANSFORMATION_CODE(x) cv::COLOR_##x
```

Comment:
Didn't notice this in the first review. Since the include files for both OpenCV 3 and 4 are in the same names, you can even do:
```c++
#if CV_MAJOR_VERSION >= 3
#include ...
#else 
#include ...
#endif 

#if CV_MAJOR_VERSION == 4
#define GET...
#else 
#define GET...
#endif
```